MapboxNavigation/CarPlayMapViewController.swift

```diff
@@ -32,22 +36,59 @@ class CarPlayMapViewController: UIViewController {
             return self.view as! NavigationMapView
         }
     }
-
-    lazy var recenterButton: CPMapButton = {
-        let recenterButton = CPMapButton { [weak self] button in
+    
+    /**
+     The map button for recentering the map view if a user action causes it to stop following the user.
+     */
+    @objc public lazy var recenterButton: CPMapButton = {
+        let recenter = CPMapButton { [weak self] button in
+            
+            self?.mapView.setUserTrackingMode(.followWithCourse, animated: true)
+            button.isHidden = true
+        }
+        let bundle = Bundle.mapboxNavigation
+        recenter.image = UIImage(named: "carplay_locate", in: bundle, compatibleWith: traitCollection)
+        return recenter
+    }()
+    
+    /**
+     The map button for zooming in the current map view.
+     */
+    @objc public lazy var zoomInButton: CPMapButton = {
```

Comment:
I’m still unsure about making these buttons into lazily computed properties, since that establishes a strict one-to-one correspondence between a button and a map view controller. I know it’s unlikely that the map would ever have multiple zoom-in buttons, for instance, but it seems like an unnecessary constraint to impose on the developer. @JThramer, do you have a preference?

Whatever the case, all the button-typed members of this class should be consistent: either they’re all lazily computed variables or they’re all factory methods without side effects.

/ref https://github.com/mapbox/mapbox-navigation-ios/pull/1962#discussion_r254898761

Reply:
@1ec5 I was the one who suggested the pattern to @vincethecoder. It seemed to me that it would be beneficial to put these in lazy var's in case the developer wanted to, for example, hide/disable the button.

Reply:
I see. It’s just that the two buttons that you’re most likely to want to show and hide, the panning buttons, are in factory methods because they require a map template as context.